play.pokemonshowdown.com/src/panel-chat.tsx

```diff
@@ -136,6 +136,130 @@ export class ChatRoom extends PSRoom {
 			this.update(null);
 			return false;
 		}
+		case 'clear': {
+			this.log?.reset();
+			this.update(null);
+			return true;
+		}
+		case 'rank':
+		case 'ranking':
+		case 'rating':
+		case 'ladder': {
+			let arg = target;
+			if (!arg) {
+				arg = PS.user.userid;
+			}
+			if (this.battle && !arg.includes(',')) {
+				arg += ", " + this.id.split('-')[1];
+			}
+
+			let targets = arg.split(',');
+			let formatTargeting = false;
+			let formats: { [key: string]: number } = {};
+			let gens: { [key: string]: number } = {};
+			for (let i = 1, len = targets.length; i < len; i++) {
+				targets[i] = $.trim(targets[i]);
+				if (targets[i].length === 4 && targets[i].substr(0, 3) === 'gen') {
+					gens[targets[i]] = 1;
+				} else {
+					formats[toID(targets[i])] = 1;
+				}
+				formatTargeting = true;
+			}
+
+			// let self = this;
+			$.get(PS.server.getActionPHP(), {
+				act: 'ladderget',
+				user: targets[0],
+			}, (data: string) => {
+				if (data.startsWith(']')) data = data.substr(1);
+				data = JSON.parse(data);
+				console.log(data);
```

Comment:
```suggestion
			PSLoginServer.query(
				'ladderget', { user: targets[0] }
			).then((data: any) => {
```

Reply:
Note that you'll need to import `PSLoginServer` as well. And get rid of the `getActionPHP` stuff.